thentos-core/src/Thentos/Types.hs

```diff
@@ -211,6 +215,38 @@ instance Aeson.FromJSON GroupNode where parseJSON = Aeson.gparseJson
 instance Aeson.ToJSON GroupNode where toJSON = Aeson.gtoJson
 
 
+-- * persona and context
+
+newtype PersonaId = PersonaId { fromPersonaId :: Integer }
+    deriving (Eq, Ord, Enum, Show, Read, Random, FromJSON, ToJSON, Typeable, Generic, FromText,
+              FromField, ToField)
+
+data Persona = Persona
+  { _personaId   :: PersonaId
+  , _personaName :: ST
+  , _personaUid  :: UserId
+  } deriving (Eq, Show, Typeable, Generic)
+
+newtype ContextId = ContextId { fromContextId :: Integer }
+    deriving (Eq, Ord, Enum, Show, Read, Random, FromJSON, ToJSON, Typeable, Generic, FromText,
+              FromField, ToField)
+
+newtype ContextName = ContextName { fromContextName :: ST }
+    deriving (Eq, Ord, Show, Read, FromJSON, ToJSON, Typeable, Generic, IsString, FromField,
+              ToField)
+
+newtype ContextDescription = ContextDescription { fromContextDescription :: ST }
+    deriving (Eq, Ord, Show, Read, FromJSON, ToJSON, Typeable, Generic, IsString, FromField,
+              ToField)
+
+data Context = Context
+  { _contextId           :: ContextId
+  , _contextOwnerService :: ServiceId
```

Comment:
I think `_contextOwnerService` is clearer but admittedly it's also a big longish. OK, I'll change it.